### PR TITLE
Fix symlink handling in project root recognition

### PR DESCRIPTION
**Pull request by aureumleonis**
Current file-to-project folder recognition doesn't work when the project is located within a symlinked directory

### CODE REVIEW

File: lsp-mode.el [hunk 1]
```diff
@@ -2864,7 +2864,7 @@ Applies on type formatting."
 
 (defun lsp-workspace-root (&optional path)
   "Find the workspace root for the current file or PATH."
-  (when-let (file-name (or path (buffer-file-name)))
+  (when-let (file-name (f-canonical (or path (buffer-file-name))))
```
Comment (yyoncho) [inline lsp-mode.el]: f-canonical will throw if called with nil

Comment (aureumleonis) [reply lsp-mode.el]: I'll update to handle the nil case

Comment (aureumleonis) [reply lsp-mode.el]: Just checked and f-ancestor-of? throws when file-name is nil so this change doesn't change ny behavior here

**Comment by yyoncho** [comment]
Thank you for contributing to `lsp-mode`!

**Comment by kelvie** [comment]
It looks like this PR: https://github.com/emacs-lsp/lsp-mode/pull/1696 broke this fix.

the new `lsp-f-canonical` does *not* resolve symlinks.

**Comment by ericdallo** [comment]
@yyoncho

**Comment by kiennq** [comment]
I wonder what's expectation for this. If a project is located inside a sym-linked folder, shouldn't it be treated as new path (not resolve symlinks)?

**Pull request merged**